crates/store/src/server/rpc_api.rs

```diff
@@ -329,6 +329,68 @@ impl rpc_server::Rpc for StoreApi {
         }))
     }
 
+    /// Returns storage map updates for the specified account within a block range.
+    ///
+    /// Supports cursor-based pagination for large storage maps.
+    #[instrument(
+        parent = None,
+        target = COMPONENT,
+        name = "store.rpc_server.sync_storage_maps",
+        skip_all,
+        level = "debug",
+        ret(level = "debug"),
+        err
+    )]
+    async fn sync_storage_maps(
+        &self,
+        request: Request<proto::rpc_store::SyncStorageMapsRequest>,
+    ) -> Result<Response<proto::rpc_store::SyncStorageMapsResponse>, Status> {
+        let request = request.into_inner();
+
+        let account_id = read_account_id(request.account_id).map_err(|e| *e)?;
+
+        if !account_id.is_public() {
+            return Err(Status::invalid_argument(format!(
+                "account with ID {account_id} is not public"
+            )));
+        }
+
+        let block_from = BlockNumber::from(request.block_from);
+        let chain_tip = self.state.latest_block_num().await;
+
+        let block_to = match request.block_to {
+            Some(block_to) => BlockNumber::from(block_to),
+            None => chain_tip,
+        };
+
+        if block_from > block_to {
+            return Err(Status::invalid_argument("block_from cannot be greater than block_to"));
+        }
```

Comment:
I think this gets handled internally as well?

That said, shouldn't this be valid and just returns zero items? But okay no strong feelings here :)

Reply:
I wasn't sure either in regards of how we would want to approach this, but seems better to try to catch it as early as possible